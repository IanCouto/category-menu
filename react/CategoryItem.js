import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'react-apollo'
import classNames from 'classnames'

import categoryQuery from './queries/categoryQuery.gql'

/**
 * Component that represents a single category displayed in the menu, also displays
 * the subcategories, if the provided category has them
 */
class CategoryItem extends Component {
  render() {
    const { data: { category } } = this.props

    const wrapperClasses = classNames('h3 pl5 pr4 vtex-category-item', {
      'show-arrow': category.hasChildren,
    })

    return (
      <div className={wrapperClasses}>
        <a href={category.href} className="db mt6 no-underline ttu black-90">
          {category.name}
        </a>

        {category.hasChildren && (
          <div className="vtex-category-sub-menu pv6 ph5 br2 br--bottom">
            <ul className="list ma0 pa0 f6">
              {category.children.map(subCategory => (
                <li key={subCategory.id} className="lh-copy">
                  <a
                    className="near-black no-underline underline-hover"
                    href={subCategory.href}
                  >
                    {subCategory.name}
                  </a>
                </li>
              ))}
            </ul>
          </div>
        )}
      </div>
    )
  }
}

CategoryItem.propTypes = {
  data: PropTypes.shape({
    category: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      href: PropTypes.string.isRequired,
      hasChildren: PropTypes.bool.isRequired,
    }),
    loading: PropTypes.bool.isRequired,
  }).isRequired,
  /** Id of the category */
  id: PropTypes.number.isRequired,
}

const options = {
  options: props => ({
    variables: {
      id: props.id,
    },
  }),
}

export default graphql(categoryQuery, options)(CategoryItem)
