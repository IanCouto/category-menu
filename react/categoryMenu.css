.itemContainer {
  max-height: 21.875rem;
}

.animation {
  left: -0.5rem;
}

@media only screen and (max-width: 40rem) {
  body.sidebarOpen {
      overflow-y: hidden;
  }
}

.container {}

.linkLevel2 {}

.linkLevel3 {}

.sidebarOpen {}

.sidebarScrim {}

.sidebar {}

.sidebarHeader {}

.sidebarContent {}

.sidebarItem {}

.sidebarContainer {}

.menuContainer {}

.sidebarItemContainer {}