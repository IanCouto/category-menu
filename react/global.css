.vtex-category-menu__item {
  border-width: 2px;
}

.vtex-category-menu__item > a {
  color: #979899;
  padding: 1.2rem 0.5rem;
}

.vtex-category-menu__item--border-blue {
  border-color: #134CD8;
}

.vtex-category-menu__item-container {
  padding-left: 8rem;
  padding-right: 8rem;
  margin-top: 0.4rem;
  max-height: 350px;
  overflow-y: auto;
}

.vtex-category-menu__item-container > div {
  width: calc(100% / 7);
}

.vtex-category-menu__item-container > div > a {
  color: #979899;
}

.vtex-category-menu__item-container > div > a:first-child {
  color: #4F4F4F;
}

.vtex-category-menu__container > span {
  border-color: #E3E4E6;
}

.vtex-category-menu__item-container > div > span,
.vtex-menu-sidebar__content > span,
.vtex-menu-sidebar__item > span {
  border-width: 0.25px;
  border-color: #E3E4E6;
  width: 90%;
  margin: 0 auto;
}

.vtex-menu-sidebar {
  height: 100vh;
  width: 85%;
}

.vtex-menu-sidebar__header > span {
  color: #585959;
}

.vtex-menu-sidebar__item path {
  fill: #979899;
}

.vtex-menu-sidebar__content {
  height: calc(100% - 2.625rem);
  overflow-y: auto;
}

.vtex-menu-sidebar__content > .vtex-menu-sidebar__item {
  color: #4F4F4F;
}

.vtex-menu-sidebar__content > .vtex-menu-sidebar__item > div > span {
  text-transform: uppercase;
} 

.vtex-menu-sidebar__item > .vtex-menu-sidebar__item {
  background-color: #F2F4F5;
  color: #727273;
  font-weight: 300;
}

.vtex-menu-sidebar__item > .vtex-menu-sidebar__item >
.vtex-menu-sidebar__item  > div > span:first-child {
  margin-left: 1rem;
}

.vtex-menu-sidebar__item > .vtex-menu-sidebar__item path {
  fill: #CACBCC;
}

.vtex-category-menu--open {
  overflow-y: hidden;
}

.vtex-menu-sidebar__foreground {
  height: 100vh;
}

@media only screen and (max-width: 49rem) {
  .vtex-category-menu {
    display: none;
  }
}
