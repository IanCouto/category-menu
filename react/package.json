{
  "name": "vtex.category-menu",
  "version": "0.0.0",
  "scripts": {
    "pretest": "yarn",
    "test": "vtex-test-tools test",
    "lint": "eslint --ext js,jsx,ts,tsx ."
  },
  "dependencies": {
    "classnames": "^2.2.6",
    "graphql": "^14.0.2",
    "jsdom": "^11.8.0",
    "ramda": "^0.26.1",
    "react-apollo": "^2.3.2",
    "react-intl": "^2.4.0"
  },
  "devDependencies": {
    "@types/classnames": "^2.2.9",
    "@types/jest": "^24.0.18",
    "@vtex/test-tools": "^0.1.2",
    "@vtex/tsconfig": "^0.1.0",
    "apollo-client": "^2.5.1"
  },
  "vtexTestTools": {
    "defaultLocale": "en"
  }
}
