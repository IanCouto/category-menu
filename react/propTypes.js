import PropTypes from 'prop-types'

function lazyFunction(f) {
  return function() {
    return f.apply(this, arguments)
  }
}

export const itemPropType = PropTypes.shape({
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  params: PropTypes.object,
  children: PropTypes.arrayOf(lazyFunction(() => itemPropType)),
})

export const categoryItemShape = PropTypes.shape({
  name: PropTypes.string.isRequired,
  slug: PropTypes.string,
  children: PropTypes.arrayOf(lazyFunction(() => itemPropType)),
})
