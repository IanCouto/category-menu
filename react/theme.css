@import url(//io.vtex.com.br/fonts/fabriga/stylesheet.css);

body,
button,
input,
optgroup,
select,
textarea,
span,
h1,
h2,
h3,
h4,
p1 {
  font-family: Fabriga, -apple-system, BlinkMacSystemFont, avenir next, avenir,
    helvetica neue, helvetica, ubuntu, roboto, noto, segoe ui, arial, sans-serif;
}

html,
body,
#render-container,
#render-provider,
#render-provider > article {
  min-height: 100%;
}

body {
  background-color: white;
}

/* Category items */

.vtex-category-menu__item {
  position: relative;
}

.vtex-category-menu__item > a {
  color: inherit;
}

.vtex-category-menu__item.show-arrow:hover::before {
  border-top-color: white;
}

.vtex-category-menu__item.show-arrow::before {
  content: ' ';
  position: absolute;
  display: inline-block;
  width: 0;
  height: 0;
  border-left: 3px solid transparent;
  border-right: 3px solid transparent;
  border-top: 3px solid black;
  top: 31px;
  left: 5px;
}

.vtex-category-menu__sub-menu {
  display: none;
  position: absolute;
  top: 100%;
  left: 0;
  width: 400px;
  min-height: 230px;
  box-shadow: 0 3px 4px 0 rgba(0, 0, 0, 0.2);
}

.vtex-category-menu__item:hover .vtex-category-menu__sub-menu {
  display: block;
}

/* Loading Bar */
.loading-container {
  position: relative;
  overflow: hidden;
}

.loading-bar {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 100%;
  z-index: 1000;
  opacity: 0.05;
  background: black;
  transform: translateX(100%);
  animation: loadbar 0.8s 0.3s infinite ease-in-out;
}

@keyframes loadbar {
  0% {
    transform: translateX(-100%);
  }
  40% {
    transform: translateX(0%);
  }
  60% {
    transform: translateX(0%);
  }
  100% {
    transform: translateX(100%);
  }
}
