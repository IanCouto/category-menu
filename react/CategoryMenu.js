import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'react-apollo'

import getCategories from './queries/categoriesQuery.gql'
import CategoryItem from './components/CategoryItem'

/**
 * Component that represents the menu containing the categories of the store
 */
class CategoryMenu extends Component {
  render() {
    const { data: { categories, loading } } = this.props

    if (loading) {
      return <div />
    }

    return (
      <div className="h3 bg-near-white">
        <nav className="flex w-two-thirds center h0">
          {categories.map(category => (
            <CategoryItem key={category.id} id={category.id} />
          ))}
        </nav>
      </div>
    )
  }
}

CategoryMenu.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    categories: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        id: PropTypes.number.isRequired,
      })
    ),
  }),
}

export default graphql(getCategories)(CategoryMenu)
