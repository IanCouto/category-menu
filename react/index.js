import PropTypes from 'prop-types'
import React, { Component, Fragment } from 'react'
import { graphql } from 'react-apollo'
import { injectIntl, intlShape } from 'react-intl'
import { IconMenu } from 'vtex.dreamstore-icons'
import { withRuntimeContext } from 'vtex.render-runtime'
import { compose, path } from 'ramda'
import classNames from 'classnames'
import { Container } from 'vtex.store-components'

import CategoryItem from './components/CategoryItem'
import SideBar from './components/SideBar'
import { categoryPropType } from './propTypes'
import getCategories from './queries/categoriesQuery.gql'

import categoryMenu from './categoryMenu.css'
import categoryMenuDisposition, {
  getMenuDispositionNames,
  getMenuDispositionValues,
} from './utils/categoryMenuDisposition'

const DEFAULT_SUBCATEGORIES_LEVELS = 1

/**
 * Component that represents the menu containing the categories of the store
 */
class CategoryMenu extends Component {
  static propTypes = {
    /** Whether to show the promotion category or not */
    showPromotionCategory: PropTypes.bool,
    /** Whether to show the gift category or not */
    showGiftCategory: PropTypes.bool,
    /** Categories query data */
    data: PropTypes.shape({
      loading: PropTypes.bool.isRequired,
      categories: PropTypes.arrayOf(categoryPropType),
    }),
    /** Set mobile mode */
    mobileMode: PropTypes.bool,
    /** Whether to show the departments category or not */
    showAllDepartments: PropTypes.bool,
    /** Whether to show subcategories or not */
    showSubcategories: PropTypes.bool,
    /** Defines the disposition of the category menu */
    menuDisposition: PropTypes.oneOf(getMenuDispositionValues()),
    /** Intl */
    intl: intlShape,
    /** Departments to be shown in the desktop mode. */
    departments: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
    })),
  }

  static defaultProps = {
    showPromotionCategory: false,
    showGiftCategory: false,
    mobileMode: false,
    showAllDepartments: true,
    showSubcategories: true,
    menuDisposition: categoryMenuDisposition.DISPLAY_LEFT.value,
    departments: [],
  }

  state = {
    sideBarVisible: false,
  }

  handleSidebarToggle = () => {
    this.setState({ sideBarVisible: !this.state.sideBarVisible })
  }

  get departments() {
    const { data: { categories = [] }, departments } = this.props
    const departmentsIds = departments.map(dept => dept.id)
    const departmentsSelected = categories.filter(category => departmentsIds.includes(category.id))

    return (departmentsSelected.length && departmentsSelected) || categories
  }

  renderSideBar() {
    const {
      intl,
      showSubcategories,
    } = this.props

    const { sideBarVisible } = this.state

    return (
      <div className={`${categoryMenu.container} ${categoryMenu.mobile}`}>
        <SideBar
          visible={sideBarVisible}
          title={intl.formatMessage({ id: 'category-menu.departments.title' })}
          departments={this.departments}
          onClose={this.handleSidebarToggle}
          showSubcategories={showSubcategories} />
        <div className="flex pa4 pointer" onClick={this.handleSidebarToggle}>
          <IconMenu size={20} />
        </div>
      </div>
    )
  }

  renderMenu() {
    const {
      data: { categories = [] },
      intl,
      showAllDepartments,
      showSubcategories,
      menuDisposition,
      runtime,
    } = this.props

    const pathName = path(['route', 'params', 'department'], runtime)

    const department = pathName ? pathName : ''

    const desktopClasses = classNames(`${categoryMenu.container} w-100 bg-base dn flex-m`, {
      'justify-start mw9': menuDisposition === categoryMenuDisposition.DISPLAY_LEFT.value,
      'justify-end mw9': menuDisposition === categoryMenuDisposition.DISPLAY_RIGHT.value,
      'justify-center': menuDisposition === categoryMenuDisposition.DISPLAY_CENTER.value,
    })

    return (
      <Container className="justify-center flex">
        <nav className={desktopClasses}>
          <ul className="pa0 list ma0 flex flex-wrap flex-row t-action overflow-hidden h3">
            {showAllDepartments &&
            <CategoryItem
              noRedirect
              menuDisposition={menuDisposition}
              subcategoryLevels={DEFAULT_SUBCATEGORIES_LEVELS + showSubcategories}
              category={{
                children: categories,
                name: intl.formatMessage({ id: 'category-menu.departments.title' }),
              }}
            />
            }
            {this.departments.map((category) => (
              <Fragment key={category.id}>
                <CategoryItem
                  menuDisposition={menuDisposition}
                  category={category}
                  subcategoryLevels={DEFAULT_SUBCATEGORIES_LEVELS + showSubcategories}
                  isCategorySelected={department === category.slug}
                />
              </Fragment>
            ))}
          </ul>
        </nav>
      </Container>
    )
  }

  render() {
    const {
      mobileMode,
    } = this.props

    return mobileMode ? this.renderSideBar() : this.renderMenu()
  }
}

export const CategoryMenuWithIntl = injectIntl(CategoryMenu)

CategoryMenuWithIntl.schema = CategoryMenu.schema = {
  title: 'editor.category-menu.title',
  description: 'editor.category-menu.description',
  type: 'object',
  properties: {
    showPromotionCategory: {
      type: 'boolean',
      title: 'editor.category-menu.show-promotion-category.title',
      default: CategoryMenu.defaultProps.showPromotionCategory,
    },
    showAllDepartments: {
      type: 'boolean',
      title: 'editor.category-menu.show-departments-category.title',
      default: CategoryMenu.defaultProps.showAllDepartments,
    },
    menuDisposition: {
      title: 'editor.category-menu.disposition-type.title',
      type: 'string',
      enum: getMenuDispositionValues(),
      enumNames: getMenuDispositionNames(),
      default: categoryMenuDisposition.DISPLAY_LEFT.value,
      isLayout: true,
    },
    showSubcategories: {
      type: 'boolean',
      title: 'editor.category-menu.show-subcategories.title',
      default: CategoryMenu.defaultProps.showSubcategories,
    },
    showGiftCategory: {
      type: 'boolean',
      title: 'editor.category-menu.show-gift-category.title',
      default: CategoryMenu.defaultProps.showGiftCategory,
    },
    departments: {
      title: 'category-menu.departments.title',
      type: 'array',
      minItems: 0,
      items: {
        title: 'editor.category-menu.departments.items.title',
        type: 'object',
        properties: {
          id: {
            title: 'editor.category-menu.departments.items.id',
            type: 'number',
          },
        },
      },
    },
  },
}

export default compose(graphql(getCategories), withRuntimeContext)(CategoryMenuWithIntl)
