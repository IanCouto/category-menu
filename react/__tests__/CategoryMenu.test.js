import React from 'react'
<<<<<<< HEAD
import { render } from '@vtex/test-tools/react'
import { CategoryMenuWithIntl } from '../index'
=======
import { render } from 'test-utils'
import { LocalCategoryMenu } from '../index'
>>>>>>> Update snapshots and remove unecessary imports

describe('CategoryMenu component', () => {
  const mockedCategories = [
    {
      id: 1,
      href: '#1',
      name: 'Category 1',
      slug: 'category-1',
      hasChildren: false,
      children: [],
    },
    {
      id: 2,
      href: '#2',
      name: 'Category 2',
      slug: 'category-2',
      hasChildren: false,
      children: [],
    },
    {
      id: 3,
      href: '#3',
      name: 'Category 3',
      slug: 'category-3',
      hasChildren: false,
      children: [],
    },
  ]

  beforeEach(() => {
    const departments = mockedCategories.map(category => ({ id: category.id }))

    wrapper = render(
      <CategoryMenuWithIntl
        data={{
          categories: mockedCategories,
          loading: false,
        }}
        departments={departments}
      />
    )
  })

  it('should be rendered', () => {
    expect(wrapper).toBeDefined()
  })

  it('should match snapshot', () => {
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it("shouldn't be able to find a `Category 4` item", () => {
    const element = wrapper.queryByText(/Category 4/)

    expect(element).toBeNull()
  })

  it('should match sidebar snapshot', () => {
    const { asFragment } = render(
      <LocalCategoryMenu
        runtime={{ hints: { mobile: true } }}
        data={{ categories: mockedCategories, loading: false }}
        departments={departments}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
