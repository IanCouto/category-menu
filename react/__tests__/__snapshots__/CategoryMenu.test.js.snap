// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryMenu component should match snapshot 1`] = `
<IntlProvider
  locale="en-US"
  messages={
    Object {
      "category-menu.departments.title": "Departments",
      "editor.category-menu.description": "A menu showing a list of the available categories on the store",
      "editor.category-menu.show-gift-category.title": "Show the gifts category",
      "editor.category-menu.show-promotion-category.title": "Show the promotion category",
      "editor.category-menu.title": "Category menu",
    }
  }
>
  <InjectIntl(CategoryMenu)
    data={
      Object {
        "categories": Array [
          Object {
            "children": Array [],
            "hasChildren": false,
            "href": "#1",
            "id": 1,
            "name": "Category 1",
            "slug": "category-1",
          },
          Object {
            "children": Array [],
            "hasChildren": false,
            "href": "#2",
            "id": 2,
            "name": "Category 2",
            "slug": "category-2",
          },
          Object {
            "children": Array [],
            "hasChildren": false,
            "href": "#3",
            "id": 3,
            "name": "Category 3",
            "slug": "category-3",
          },
        ],
        "loading": false,
      }
    }
  >
    <CategoryMenu
      data={
        Object {
          "categories": Array [
            Object {
              "children": Array [],
              "hasChildren": false,
              "href": "#1",
              "id": 1,
              "name": "Category 1",
              "slug": "category-1",
            },
            Object {
              "children": Array [],
              "hasChildren": false,
              "href": "#2",
              "id": 2,
              "name": "Category 2",
              "slug": "category-2",
            },
            Object {
              "children": Array [],
              "hasChildren": false,
              "href": "#3",
              "id": 3,
              "name": "Category 3",
              "slug": "category-3",
            },
          ],
          "loading": false,
        }
      }
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "formatDate": [Function],
          "formatHTMLMessage": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatPlural": [Function],
          "formatRelative": [Function],
          "formatTime": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralFormat": [Function],
            "getRelativeFormat": [Function],
          },
          "locale": "en-US",
          "messages": Object {
            "category-menu.departments.title": "Departments",
            "editor.category-menu.description": "A menu showing a list of the available categories on the store",
            "editor.category-menu.show-gift-category.title": "Show the gifts category",
            "editor.category-menu.show-promotion-category.title": "Show the promotion category",
            "editor.category-menu.title": "Category menu",
          },
          "now": [Function],
          "textComponent": "span",
        }
      }
      mobileMode={false}
      showGiftCategory={false}
      showPromotionCategory={false}
    >
      <div
        className="vtex-category-menu flex justify-center items-center bg-white"
      >
        <div
          className="vtex-category-menu__container w-100 h-100 flex justify-between items-center f6 overflow-hidden"
        >
          <CategoryItem
            category={
              Object {
                "children": Array [
                  Object {
                    "children": Array [],
                    "hasChildren": false,
                    "href": "#1",
                    "id": 1,
                    "name": "Category 1",
                    "slug": "category-1",
                  },
                  Object {
                    "children": Array [],
                    "hasChildren": false,
                    "href": "#2",
                    "id": 2,
                    "name": "Category 2",
                    "slug": "category-2",
                  },
                  Object {
                    "children": Array [],
                    "hasChildren": false,
                    "href": "#3",
                    "id": 3,
                    "name": "Category 3",
                    "slug": "category-3",
                  },
                ],
                "name": "Departments",
              }
            }
            noRedirect={true}
            widthPercent={25}
          >
            <div
              className="vtex-category-menu__item flex justify-center items-center bb b--transparent"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "width": "25%",
                }
              }
            >
              <a
                className="w-100 no-underline f6 outline-0 db tc truncate"
                href="#"
              >
                DEPARTMENTS
              </a>
              <div
                className="absolute w-100 left-0"
                style={
                  Object {
                    "display": "none",
                    "top": undefined,
                  }
                }
              >
                <ItemContainer
                  categories={
                    Array [
                      Object {
                        "children": Array [],
                        "hasChildren": false,
                        "href": "#1",
                        "id": 1,
                        "name": "Category 1",
                        "slug": "category-1",
                      },
                      Object {
                        "children": Array [],
                        "hasChildren": false,
                        "href": "#2",
                        "id": 2,
                        "name": "Category 2",
                        "slug": "category-2",
                      },
                      Object {
                        "children": Array [],
                        "hasChildren": false,
                        "href": "#3",
                        "id": 3,
                        "name": "Category 3",
                        "slug": "category-3",
                      },
                    ]
                  }
                >
                  <div
                    className="vtex-category-menu__item-container w-100 bg-white shadow-5 pb2 dib overflow-y-auto"
                  >
                    <div
                      className="fl db pa2"
                      key="1"
                    >
                      <Link
                        className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                        page="store/department"
                        params={
                          Object {
                            "department": "category-1",
                          }
                        }
                      >
                        <a
                          className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                          href="/category-1/s"
                        >
                          CATEGORY 1
                        </a>
                      </Link>
                    </div>
                    <div
                      className="fl db pa2"
                      key="2"
                    >
                      <Link
                        className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                        page="store/department"
                        params={
                          Object {
                            "department": "category-2",
                          }
                        }
                      >
                        <a
                          className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                          href="/category-2/s"
                        >
                          CATEGORY 2
                        </a>
                      </Link>
                    </div>
                    <div
                      className="fl db pa2"
                      key="3"
                    >
                      <Link
                        className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                        page="store/department"
                        params={
                          Object {
                            "department": "category-3",
                          }
                        }
                      >
                        <a
                          className="vtex-category-menu__link-level-2 db f6 fw4 no-underline pa4 outline-0 tl truncate"
                          href="/category-3/s"
                        >
                          CATEGORY 3
                        </a>
                      </Link>
                    </div>
                  </div>
                </ItemContainer>
              </div>
            </div>
          </CategoryItem>
          <span
            className="br bw1 h1"
          />
          <CategoryItem
            category={
              Object {
                "children": Array [],
                "hasChildren": false,
                "href": "#1",
                "id": 1,
                "name": "Category 1",
                "slug": "category-1",
              }
            }
            widthPercent={25}
          >
            <div
              className="vtex-category-menu__item flex justify-center items-center bb b--transparent"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "width": "25%",
                }
              }
            >
              <Link
                className="w-100 no-underline f6 outline-0 db tc truncate"
                page="store/department"
                params={
                  Object {
                    "department": "category-1",
                  }
                }
              >
                <a
                  className="w-100 no-underline f6 outline-0 db tc truncate"
                  href="/category-1/s"
                >
                  CATEGORY 1
                </a>
              </Link>
            </div>
          </CategoryItem>
          <span
            className="br bw1 h1"
          />
          <CategoryItem
            category={
              Object {
                "children": Array [],
                "hasChildren": false,
                "href": "#2",
                "id": 2,
                "name": "Category 2",
                "slug": "category-2",
              }
            }
            widthPercent={25}
          >
            <div
              className="vtex-category-menu__item flex justify-center items-center bb b--transparent"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "width": "25%",
                }
              }
            >
              <Link
                className="w-100 no-underline f6 outline-0 db tc truncate"
                page="store/department"
                params={
                  Object {
                    "department": "category-2",
                  }
                }
              >
                <a
                  className="w-100 no-underline f6 outline-0 db tc truncate"
                  href="/category-2/s"
                >
                  CATEGORY 2
                </a>
              </Link>
            </div>
          </CategoryItem>
          <span
            className="br bw1 h1"
          />
          <CategoryItem
            category={
              Object {
                "children": Array [],
                "hasChildren": false,
                "href": "#3",
                "id": 3,
                "name": "Category 3",
                "slug": "category-3",
              }
            }
            widthPercent={25}
          >
            <div
              className="vtex-category-menu__item flex justify-center items-center bb b--transparent"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              style={
                Object {
                  "width": "25%",
                }
              }
            >
              <Link
                className="w-100 no-underline f6 outline-0 db tc truncate"
                page="store/department"
                params={
                  Object {
                    "department": "category-3",
                  }
                }
              >
                <a
                  className="w-100 no-underline f6 outline-0 db tc truncate"
                  href="/category-3/s"
                >
                  CATEGORY 3
                </a>
              </Link>
            </div>
          </CategoryItem>
        </div>
      </div>
    </CategoryMenu>
  </InjectIntl(CategoryMenu)>
</IntlProvider>
`;
